src/v2/mod.rs

```diff
@@ -1057,6 +1151,22 @@ impl IntoBlockIdentifier for RelativeBlockHeight {
     fn into_block_identifier(self) -> BlockIdentifier { BlockIdentifier::RelativeHeight(self) }
 }
 
+pub trait IntoEpochIdentifier {
+    fn into_epoch_identifier(self) -> EpochIdentifier;
+}
+
+impl IntoEpochIdentifier for EpochIdentifier {
+    fn into_epoch_identifier(self) -> EpochIdentifier { self }
+}
+
+impl IntoEpochIdentifier for RelativeEpoch {
+    fn into_epoch_identifier(self) -> EpochIdentifier { EpochIdentifier::Relative(self) }
+}
+
+impl IntoEpochIdentifier for BlockIdentifier {
+    fn into_epoch_identifier(self) -> EpochIdentifier { EpochIdentifier::Block(self) }
+}
```

Comment:
With only these instances, you would get the same behaviour if the type of `get_winning_bakers_epoch` would be

```rust
get_winning_bakers_epoch(..., ei: impl Into<EpochIdentifier>)
```

Reply:
And then you don't need the trait.